mbuild/compound.py

```diff
@@ -1138,7 +1138,159 @@ def _kick(self):
+            _minimize_energy_openmm
+
+        """
+        tmp_dir = tempfile.mkdtemp()
+        original = clone(self)
+        self._kick()
+        self.save(os.path.join(tmp_dir,'un-minimized.mol2'))
+        extension = os.path.splitext(forcefield)[-1]
+        if extension == '.xml':
+            self._minimize_energy_openmm(tmp_dir, forcefield=forcefield,
+                    steps=steps)
+
+        else:
+            self._minimize_energy_openbabel(tmp_dir, forcefield=forcefield,
+                    steps=steps)
+
+
+        self.update_coordinates(os.path.join(tmp_dir, 'minimized.pdb'))
+
+    def _minimize_energy_openmm(self, tmp_dir, forcefield="", steps=1000,
+            scale_bonds=1, scale_angles=1, scale_torsions=0.5,
+            scale_nonbonded=0.75):
+        """ Perform energy minimization using OpenMM
+
+        Converts an mBuild Compound to a Parmed Structure,
+        applies a forcefield, and creates an OpenMM System.
+
+        Parameters
+        ----------
+        forcefield = str
+            Path to forcefield xml file
+        steps = int
+            Number of energy minimization iterations
+        scale_bonds = float
+            Scales the bond force constant (1 is completely on)
+        scale_angles = float
+            Scales the angle force constant (1 is completely on)
+        scale_torsions = float
+            Scales the torsional force constants (1 is completely on)
+        scale_nonbonded = float
+            Scales epsilon (1 is completely on)
+
+
+        Notes
+        -----
+        Assumes a particular organization for the force groups
+        (HarmonicBondForce, HarmonicAngleForce, RBTorsionForce, NonBondedForce)
+
+
+        """
+        from foyer import Forcefield
+        to_parmed = self.to_parmed()
+        ff = Forcefield(forcefield_files=[forcefield])
+        to_parmed = ff.apply(to_parmed)
+
+
+        #system = to_parmed.createSystem(
+                #nonbondedMethod=PME,
+                #nonbondedCutoff=0.1*nanometer,constraints=None)
+        from simtk.openmm.app.simulation import Simulation
+        from simtk.openmm.app.pdbreporter import PDBReporter
+        from simtk.openmm.openmm import LangevinIntegrator
+        import simtk.unit as u
+
+        system = to_parmed.createSystem()
+        integrator = LangevinIntegrator(298*u.kelvin, 1/u.picosecond, 
+                0.002*u.picoseconds)
+        simulation = Simulation(to_parmed.topology, system, integrator)
+
+
+        # Reduce the force constants to prevent blowing up simulation
+        # And also relax energy barriers (especially useful for torsions)
+        all_forces = system.getForces()
+        bond_forces = all_forces[0]
+        angle_forces = all_forces[1]
+        torsion_forces = all_forces[2]
+        nb_forces = all_forces[3]
+
+        # Scaling HarmonicBondForces
+        for bond_index in range(bond_forces.getNumBonds()):
+            old_bond_parameters=bond_forces.getBondParameters(bond_index)
```

Comment:
If possible, this would be easier to read if the parameters were unpacked into separate variables, i.e.
```
atom1, atom2, r0, k = bond_forces.getBondParameters(bond_index)
```
Ditto for the angle, dihedral, and pair parameters.